manifests/kernel_upgrade.pp

```diff
@@ -127,8 +149,50 @@
     } else {
       $minute = $update_minute
     }
+    # CREATE update_months STRING
+    $update_months_string = join($update_months, ',')
+    $months =
+      regsubst(
+        regsubst(
+          regsubst(
+            regsubst(
+              regsubst(
+                regsubst(
+                  regsubst(
+                    regsubst(
+                      regsubst(
+                        regsubst(
+                          regsubst(
+                            regsubst(
+                              regsubst(
+                                regsubst(
+                                  regsubst(
+                                    regsubst(
+                                      regsubst(
+                                        regsubst(
+                                          regsubst($update_months_string, ',', ', ', 'G'),
+                                          '\*/2', 'Feb, Apr, Jun, Aug, Oct, Dec'
+                                        ), '\*/3', 'Mar, Jun, Sep, Dec'
+                                      ), '\*/4', 'Apr, Aug, Dec'
+                                    ), '\*/5', 'May & Oct'
+                                  ), '\*/6', 'Jun & Dec'
+                                ), '\*', 'each month'
+                              ), '12', 'Dec'
+                            ), '11', 'Nov'
+                          ), '10', 'Oct'
+                        ), '9', 'Sep',
+                      ), '8', 'Aug'
+                    ), '7', 'Jul'
+                  ), '6', 'Jun'
+                ), '5', 'May'
+              ), '4', 'Apr'
+            ), '3', 'Mar'
+          ), '2', 'Feb'
+        ), '1', 'Jan'
+      )
```

Comment:
I know this is friggin ugly. It works, but would love to have a more manageable way to convert the $update_months into a pretty string for the motd notice.

Any suggestions for how to improve this?

Reply:
discussed privately and resolution included in PR comment below.